### PR TITLE
feature: add image uri for lambda source

### PR DESCRIPTION
**Pull request by Plork**
add attribute for ECR image URI containing the function's deployment package.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ locals {
 module "lambda_role" {
   count = length(compact([var.role_arn])) == 0 ? 1 : 0
 
-  source                = "github.com/schubergphilis/terraform-aws-mcaf-role?ref=v0.3.3"
+  source                = "github.com/schubergphilis/terraform-aws-mcaf-role?ref=v0.4.0"
```
Comment (marwinbaumannsbp) [inline main.tf:16]: Registry source?

File: variables.tf [hunk 2]
```diff
@@ -81,6 +81,17 @@ variable "handler" {
   description = "The function entrypoint in your code"
 }
 
+variable "image_uri" {
+  type        = string
+  default     = null
+  description = "ECR image URI containing the function's deployment package. (must be internal ECR URI to the organiusation)"
```
Comment (marwinbaumannsbp) [inline variables.tf:87]:
```suggestion
  description = "ECR image URI containing the function's deployment package. (must be internal ECR URI)"
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ Terraform module to create an AWS Lambda function.
 
 IMPORTANT: We do not pin modules to versions in our examples. We highly recommend that in your code you pin the version to the exact version you are using so that your infrastructure remains stable.
 
+IMPORTANT: exactly one of filename, image_uri, or s3_bucket must be specified when using the module.
```
Comment (marwinbaumannsbp) [inline README.md:7]: We could do cross variable validatioin which is now availble in v1.9. What do you think?

Comment (Plork) [reply README.md:7]: trust me I tried:

```
│ Error: Cycle: module.lambda.var.filename (validation), module.lambda.var.s3_bucket (validation), module.lambda.var.image_uri (validation)
│

Comment (Plork) [reply README.md:7]: this does not work or maybe I am doing something wrong.

```hcl
variable "image_uri" {
  type        = string
  description = "The URI of the image"
  default     = null

  validation {
    condition = (
      !(var.image_uri != null && (var.filename != null || var.s3_bucket != null))
    )
    error_message = "You can specify only one of image_uri, filename, or s3_bucket."
  }
}

variable "filename" {
  type        = string
  description = "The filename"
  default     = null

  validation {
    condition = (
      !(var.filename != null && (var.image_uri != null || var.s3_bucket != null))
    )
    error_message = "You can specify only one of image_uri, filename, or s3_bucket."
  }
}

variable "s3_bucket" {
  type        = string
  description = "The S3 bucket name"
  default     = null

  validation {
    condition = (
      !(var.s3_bucket != null && (var.image_uri != null || var.filename != null))
    )
    error_message = "You can specify only one of image_uri, filename, or s3_bucket."
  }
}
```

**Comment by marwinbaumannsbp** [comment]
stale

**Pull request closed without merging**